lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -622,20 +622,31 @@ def add_mpi_data_for_network_tokenization_card(post, payment, options)
 
         post[:mpiData] = {}
         post[:mpiData][:authenticationResponse] = 'Y'
-        post[:mpiData][:cavv] = payment.payment_cryptogram
+        if options[:switch_cryptogram_mapping_nt]
+          post[:mpiData][:tokenAuthenticationVerificationValue] = payment.payment_cryptogram
```

Comment:
I'm not against putting it behind a flag as well, but at some point we will want to remove the flag, and I think we will need both mappings at that point.

Reply:
On previous tests we got some error some in order to avoid a double deploy while we assure the proper behavior we are going to use the LD approach. @almalee24 could give you some additional info on that

Reply:
Yeah last time the told us to change this field but it failed when it went out which is why we are doing it with a LD flag. Just to be extra cautious and it'll be easier to turn off it sometimes goes wrong.

Reply:
But we should update it to be `options[:switch_cryptogram_mapping_nt] && payment.is_a?(NetworkTokenizationCreditCard)`

Reply:
Yeah that sounds good, if we use the flag and also check the payment type, then if it works we can remove the flag, and keep the payment type to have the same behavior.

Reply:
after running tests, the switch in cryptogram mapping causes an error on mpidata so this latest update keeps the adapter correct

Reply:
We should probably talk to adyen about this. This issue started because there were failures that adyen said was because the cryptogram was sent in the `cavv` instead of the `tokenAuthenticationVerificationValue` field. 

Need to make sure we understand what exactly the failures are in the remote tests and how we should proceed with these fields long term.

Reply:
So is there a way to contact adyen support? as far as I know is not easy to get in touch with then.

Reply:
I'm not sure. Did you see my comment about trying to have these updates only apply to NetworkTokens, and not ApplePay and GooglePay? It seems that if we can keep the apple pay and google pay integration the same, and only update the fields we are sending for NetworkTokens it may solve the problems you are seeing in the tests.

Reply:
Noted! Done!

Reply:
@DustinHaefele can you take a look at this PR again since you have requested changes?